src/main/java/Rabbit/util/TaskList.java

```diff
@@ -0,0 +1,203 @@
+package Rabbit.util;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import Rabbit.RabbitException.FindFormatException;
+import Rabbit.Task.Task;
+import Rabbit.Task.Event;
+import Rabbit.Task.Deadline;
+import Rabbit.Task.Todo;
+import Rabbit.RabbitException.AddToListException;
+import Rabbit.RabbitException.MarkUnmarkException;
+import Rabbit.RabbitException.DeleteException;
+
+
+public class TaskList {
+    private ArrayList<Task> list;
+    public enum TaskType {
+        TODO, DEADLINE, EVENT;
+    }
+
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Returns tasks in the list that match
+     * the searching keyword.
+     *
+     * @param input the searching keyword.
+     * @return the tasks matching the searching keyword.
+     * @throws FindFormatException
+     */
+    public String find(String input) throws FindFormatException {
+        try {
+            String keyword = Parser.parseFind(input, this);
+            if (keyword.equals(" ")) {
+                throw new FindFormatException();
+            }
+            String output = "";
+            for (int i = 0; i < this.list.size(); i++) {
+                String content = this.list.get(i).getContent();
+
+                if (content.length() < keyword.length()) {
+                    continue;
+                }
+
+                for (int j = 0; j < content.length() - keyword.length() + 1; j++) {
+                    if (content.substring(j, j + keyword.length()).equals(keyword)) {
+                        output += list.get(i).toString();
+                    }
+                }
+            }
+            return output.length() == 0 ? "I can't find a matching task." : output;
+        } catch (FindFormatException e) {
+            throw e;
+        }
+    }
+    
+    /**
+     *  Adds the input lines the user types into
+     *  a list with a size no more than 100.
+     *
+     * @param task the type of the task that is to be added.
+     * @param input the content (and the time) of the task the user inputs.
+     * @return the content of the task.
+     */
+    public String addToList(TaskType task, String input) throws AddToListException {
+        if (this.list.size() == 100) {
+            // throws an exception when there are already 100 lines in
+            // the list when the user is trying to input a new line.
```

Comment:
I find your JavaDoc descriptions and in-code comments really useful in navigating the purpose of methods and fields! It seems to adhere very well to the Guideline of **"Comment minimally, but sufficiently"** given in the [Code Quality Guidelines](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html).